rollup.config.js

```diff
@@ -1,32 +1,69 @@
-import commonjs from 'rollup-plugin-commonjs';
-import nodeResolve from 'rollup-plugin-node-resolve';
-import nodeBuiltins from 'rollup-plugin-node-builtins';
-import babel from 'rollup-plugin-babel';
-import sourcemaps from 'rollup-plugin-sourcemaps';
-import { uglify } from "rollup-plugin-uglify";
-import banner from 'rollup-plugin-banner';
-import genHeader from './lib/header';
+import commonjs from 'rollup-plugin-commonjs'
+import resolve from 'rollup-plugin-node-resolve'
+import babel from 'rollup-plugin-babel'
+import typescript from 'rollup-plugin-typescript2'
+import banner from 'rollup-plugin-banner'
+import minify from 'rollup-plugin-babel-minify'
+import genHeader from './lib/header'
 
-const genSourceMap = false;
+// TypeScript because Babel transpiles modules in isolation, therefore no type reexports.
+// CommonJS because Babel is not 100 % ESM.
 
-export default [{
-	input: 'index.js',
-	output: {
-		file: 'dist/vis-network.min.js',
-		name: 'vis',
-		exports: 'named',
-		format: 'umd',
-		sourcemap: genSourceMap
-	},
-	plugins: [
-		commonjs(),
-		nodeBuiltins(),
-		nodeResolve(),
-		babel({ runtimeHelpers: true }),
-		sourcemaps(),
-		uglify({
-			sourcemap: genSourceMap
-		}),
-		banner(genHeader('network'))
-	]
-}]
+const plugins = [
+  resolve({
+    extensions: ['.ts', '.js', '.json'],
+  }),
+  typescript({
+    tsconfig: 'tsconfig.code.json',
+  }),
+  commonjs(),
+  babel({
+    extensions: ['.ts', '.js'],
+    runtimeHelpers: true,
+  }),
+  banner(genHeader('network')),
+]
+const minPlugins = [
+  ...plugins.slice(0, 4),
+  minify({ comments: false }),
+  ...plugins.slice(4)
+]
+
+export default [
+  {
+    input: 'lib/index.ts',
+    output: {
+      file: 'dist/esm.js',
+      format: 'esm',
+    },
+    plugins,
+  },
+  {
+    input: 'lib/index.ts',
+    output: {
+      file: 'dist/umd.js',
+      format: 'umd',
+      exports: 'named',
+      name: 'vis',
+    },
+    plugins,
+  },
+  {
+    input: 'lib/index.ts',
+    output: {
+      file: 'dist/esm.min.js',
```

Comment:
see above..

Reply:
The files are too different, Git just doesn't see it.